tests/run/i17255/J.java

```diff
@@ -0,0 +1,13 @@
+package p;
+
+public class J {
+	public static J j = new J();
+	public static p.J f() {
+		return p.J.j;
+	}
+	public static p.Module$ module() {
```

Comment:
should also test the case without a package prefix, because this activates a different code path (findRef), and is actually correctly typed as (p.Module.type).
```suggestion
	public static Module$ module2() {
		return p.Module$.MODULE$;
	}
	public static p.Module$ module() {
```

Reply:
also the tree ends up being 
```
   101:       DEFDEF(12) 20 [module2]
   104:         EMPTYCLAUSE
   105:         IDENTtpt 17 [Module$]
   107:           TYPEREF 19 [Module[ModuleClass]]
   109:             SHAREDtype 12
   111:         ELIDED
   112:           SHAREDtype 107
   114:         STATIC
```
where `105` would ideally become `IDENTtpt 19 [Module[ModuleClass]`